code/datums/components/crafting/recipes.dm

```diff
@@ -42,6 +49,18 @@
 	if(tool_paths)
 		tool_paths = string_list(tool_paths)
 
+/datum/crafting_recipe/stack/New(obj/item/stack/material, datum/stack_recipe/stack_recipe)
+	if(!material || !stack_recipe || !stack_recipe.result_type)
+		return
```

Comment:
throw an exception here instead of just silently failing.
If a material is not being passed in or a stack recipe is not being passed in, the code should definitely be throwing an error.

Reply:
There are stack recipes that are just groups of other recipes. 
Like fancy boxes that is not a recipe, but a holder for other ones.

But I guess I should check for them in a separate statement. And indeed throw an exception here.

Reply:
you mean abstract types?
You can solve that by having an abstract_type var that's set to the abstract type
Then a simple `if(type == abstract_type)` check will let you determine if it's the abstract type or not

Reply:
It's a separate datum: `stack_recipe_list`. I will add a check for this type.